adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -544,30 +560,37 @@ public final void uploadEventMessage(final RoutingContext ctx, final String tena
                 payload,
                 contentType,
                 getEventSender(tenant),
-                EventConstants.EVENT_ENDPOINT);
+                EndpointType.EVENT);
     }
 
-    private void doUploadMessage(final RoutingContext ctx, final String tenant, final String deviceId,
-            final Buffer payload, final String contentType, final Future<MessageSender> senderTracker, final String endpointName) {
+    private void doUploadMessage(
+            final RoutingContext ctx,
+            final String tenant,
+            final String deviceId,
+            final Buffer payload,
+            final String contentType,
+            final Future<MessageSender> senderTracker,
+            final EndpointType endpoint) {
 
         if (!isPayloadOfIndicatedType(payload, contentType)) {
             HttpUtils.badRequest(ctx, String.format("content type [%s] does not match payload", contentType));
         } else {
             final String qosHeaderValue = ctx.request().getHeader(Constants.HEADER_QOS_LEVEL);
-            final Integer qos = getQoSLevel(qosHeaderValue);
-            if (qos != null && qos == HEADER_QOS_INVALID) {
+            final MetricsTags.QoS qos = getQoSLevel(qosHeaderValue);
+            if (MetricsTags.QoS.UNKNOWN.equals(qos)) {
```

Comment:
As enums are guaranteed to have only one instance per literal, you can do a comparison of the object reference, which should be faster than calling the equals method.

Reply:
will do